### PR TITLE
upgrade to new api-version for `Azure.ResourceManager.Storage`

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs [hunk 1]
```diff
@@ -2314,6 +2379,19 @@ public partial class StorageUsage
...
+        public TableAccessPolicy(string permission) { }
+        public System.DateTimeOffset? ExpiryOn { get { throw null; } set { } }
+        public string Permission { get { throw null; } set { } }
+        public System.DateTimeOffset? StartOn { get { throw null; } set { } }
+    }
+    public partial class TableSignedIdentifier
+    {
+        public TableSignedIdentifier(string id) { }
+        public Azure.ResourceManager.Storage.Models.TableAccessPolicy AccessPolicy { get { throw null; } set { } }
+        public string Id { get { throw null; } set { } }
```
Comment (ArthurMa1978) [inline sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:2393]: Is this a `ResourceIdentifier` ?

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs [hunk 2]
```diff
@@ -2186,6 +2243,14 @@ public partial class StorageAccountRegenerateKeyContent
         public StorageAccountRegenerateKeyContent(string keyName) { }
         public string KeyName { get { throw null; } }
     }
+    public partial class StorageAccountSkuConversionStatus
+    {
+        public StorageAccountSkuConversionStatus() { }
+        public string EndTime { get { throw null; } }
```
Comment (ArthurMa1978) [inline sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:2249]: Is this a `DateTimeOffset` type

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/storage/Azure.ResourceManager.Storage/src/autorest.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ azure-arm: true
 csharp: true
 namespace: Azure.ResourceManager.Storage
 tag: package-2021-09
-require: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/4124b7c2773a714303299f0cfd742b0d26d3bb5d/specification/storage/resource-manager/readme.md
+require: C:\Users\dapzhang\Documents\workspace\azure-rest-api-specs/specification/storage/resource-manager/readme.md
```
Comment (Yao725) [inline sdk/storage/Azure.ResourceManager.Storage/src/autorest.md:10]: Please change back this link.

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs (hunk 2 above)
Comment (ArcturusZhang) [reply sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:2249]: We will have an api-view review later

**Pull request merged**